src/sbvr-api/permissions.ts

```diff
@@ -684,6 +685,200 @@ const getAlias = (name: string) => {
+					}
+
+					const targetResource =
+						abstractSqlModel.tables[possibleTargetResourceName];
+					const targetResourceEscaped = sqlNameToODataName(
+						targetResource?.name ?? possibleTargetResourceName,
+					);
+
+					// This is either a translated or bypassed resource we don't
+					// mess with these
+					if (targetResourceEscaped.includes('$')) {
+						return;
+					}
+
+					let foundCanAccessLink = false;
+
+					try {
+						const odata = memoizedParseOdata(`/${targetResourceEscaped}`);
+
+						const collapsedPermissionFilters = buildODataPermission(
+							permissionsLookup,
+							methodPermissions.GET,
+							vocabulary,
+							targetResourceEscaped,
+							odata,
+						);
+
+						_.set(
+							odata,
+							['tree', 'options', '$filter'],
+							collapsedPermissionFilters,
+						);
+
+						const canAccessFunction: ResourceFunction = function(
+							property: AnyObject,
+						) {
+							// remove method property so that we won't loop back here again at this point
+							delete property.method;
+
+							if (!this.defaultResource) {
+								throw new Error(`No resource selected in AST.`);
+							}
+
+							const targetResourceAST = this.NavigateResources(
+								this.defaultResource,
+								property.name,
+							);
+
+							const targetResourceName = sqlNameToODataName(
+								targetResourceAST.resource.name,
+							);
+							const currentResourceName = sqlNameToODataName(
+								this.defaultResource.name,
+							);
+
+							if (
+								currentResourceName === targetResourceEscaped &&
+								targetResourceName === escapedName
+							) {
+								foundCanAccessLink = true;
+							}
+							// return a true expression to not select the relationship, which might be virtual
+							// this should be a boolean expression, but needs to be a subquery in case it
+							// is wrapped in an `or` or `and`
+							return ['Equals', ['Boolean', true], ['Boolean', true]];
+						};
+
+						// We need execute the
+						const odata2AbstractSQL = new OData2AbstractSQL(
+							originalAbstractSQLModel,
+							{
+								canAccess: canAccessFunction,
+							},
+						);
+
+						try {
+							odata2AbstractSQL.match(
+								odata.tree,
+								'GET',
+								[],
+								odata.binds.length,
+							);
+						} catch (e) {
+							throw new ODataParser.SyntaxError(e);
+						}
+
+						if (foundCanAccessLink) {
+							// store the resource name as it was with a $bypass
+							// suffix in this relationship, this means that the
+							// query generator will use the plain resource instead
+							// of the filtered resource.
+							arr[1][0] = `${possibleTargetResourceName}$bypass`;
+						}
+					} catch (e) {
+						if (e === constrainedPermissionError) {
+							// ignore
+							return;
+						}
+
+						if (e instanceof ODataParser.SyntaxError) {
+							// we can ignore these, because this will occur
+							// often until we have full support for translations
+							// in pine
```

Comment:
Why does it occur for translations?

Reply:
It will fail to find resources like `application$resin$bypass`, because they might not exist in the model.